### PR TITLE
Reorganize the Preferences UI -- refactored

### PR DESCRIPTION
**Pull request by dirkhh**
### Describe the pull request:

- [ ] Bug fix
- [x] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

this is a refactored version of #2436 

@willemferguson let me know if this works for you

### CODE REVIEW

File: desktop-widgets/preferences/preferences_equipment.cpp [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	ui->display_unused_tanks->setChecked(prefs.display_unused_tanks);
+	ui->default_cylinder->clear();
+	for (int i = 0; i < MAX_TANK_INFO && tank_info[i].name != NULL; i++) {
+		ui->default_cylinder->addItem(tank_info[i].name);
+		if (qPrefEquipment::default_cylinder() == tank_info[i].name)
+			ui->default_cylinder->setCurrentIndex(i);
+	}
+}
+
+#include <QDebug>
```
Comment (bstoeger) [inline desktop-widgets/preferences/preferences_equipment.cpp]: Why?

File: desktop-widgets/preferences/preferences_media.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	ui->ffmpegFile->setEnabled(qPrefMedia::extract_video_thumbnails());
+
+	ui->extractVideoThumbnails->setChecked(qPrefMedia::extract_video_thumbnails());
+	ui->videoThumbnailPosition->setValue(qPrefMedia::extract_video_thumbnails_position());
+	ui->ffmpegExecutable->setText(qPrefMedia::ffmpeg_executable());
+
+	ui->auto_recalculate_thumbnails->setChecked(prefs.auto_recalculate_thumbnails);
+}
+
+#include <QDebug>
```
Comment (bstoeger) [inline desktop-widgets/preferences/preferences_media.cpp]: Why?

**Comment by willemferguson** [comment]
I went through the code and the commented-out code appears in the 6th and the 8th commits (54759 and 88d5e) in the two files qPrefGneral.cpp and TestQPrefTechnicalDetails.cpp. 

As Berhold pointed out, there are QDebug #includes in preferences_equipment.cpp and preferences_media.cpp: a remnant of Ferguson's debugging during developing.

For the rest everything appears ok to me.

There are a few issues I would like to deal with:
I think Berthold is the guilty party for creating the code that makes this code sharable with the mobile code base (this is excellent, given my contention that the future lies in mobile apps). Therefore it would probably be extremely useful if Berthold were prepared to carefully look at this PR. I cannot build for mobile at the moment therefore I cannot test mobile compatibility. I did add the additional qPrefxx files to packaging/ios/subsurface-mobile.pro but I do not know whether this is sufficient.

After dealing with this PR the code for the three test programs should be looked at carefully. I am convinced that code is not accurate when dealing with QSignalSpy monitoring of the non-boolean variables.

I have already identified some enhancements on the Mac screen display of some of the widgets (based on Dirk's screenshots) but I will get to that after this PR has been dealt with.

Please let me know what I can do myself with respect to dealing with this PR. I hate Dirk taking some responsibility for the issues I created in some files. Dirk, I would appreciate it we could decide on take-home messages for me. It looks like, in this case, I should have been much, much more careful with commenting out code and not carry them into commits. If there are any other take-home messages for me I would appreciate them.

File: desktop-widgets/preferences/preferences_equipment.cpp (hunk 1 above)
Comment (dirkhh) [reply desktop-widgets/preferences/preferences_equipment.cpp]: almost certainly a merge error on my part. Will remove.

File: desktop-widgets/preferences/preferences_media.cpp (hunk 2 above)
Comment (dirkhh) [reply desktop-widgets/preferences/preferences_media.cpp]: ditto

**Comment by dirkhh** [comment]
looks like I still have a little more work to do.
I fixed the commit message subject lines - those should be consistent on a series (where reasonable) and give a super brief description of the main thing a commit does. Just calling every commit "Preference UI rework" or something like that doesn't help me when I look at the commit list.
The key message for you @willemferguson is to either become comfortable with `git rebase -i` or to be very careful what you actually commit. `git add -p` can help a lot with that. Think about the commits from a reader's perspective. And think about the series as something that should solve one problem, broken down into reasonable chunks.

**Comment by dirkhh** [comment]
I completely dropped the ball here. Too much travel, too little time.
I assume this is still the correct branch to clean up, @willemferguson ?
My sincere apologies

**Comment by dirkhh** [comment]
OK, here's another attempt with better cleanup on the way, and rebased on current master. Let's see what GitHub Actions think about this.

**Comment by willemferguson** [comment]
I checked the code carefully and it looks perfect.

**Pull request merged**